spec/requests/clients_spec.rb

```diff
@@ -1,77 +1,50 @@
 require 'rails_helper'
 
-RSpec.describe 'Clients', type: :request, elasticsearch: true  do
-  let!(:provider) { build(:provider) }
-  let!(:clients)  { build_list(:client, 10, provider_id: provider.symbol) }
-  let!(:client) { clients.first }
+describe 'Clients', type: :request, elasticsearch: true  do
+  let(:parameters) { ActionController::Parameters.new(id: "abc.def", symbol: "ABC.DEF", name: "Test", contact_name: "Josiah Carberry", contact_email: "josiah@example.org", created: Time.zone.now) }
+  let!(:client) { Client.create(parameters.permit(Client.safe_params)) }
   let(:params) do
     { "data" => { "type" => "clients",
                   "attributes" => {
-                    "symbol" => provider.symbol+".IMPERIAL",
+                    "symbol" => client.symbol,
                     "name" => "Imperial College",
                     "contact-name" => "Madonna",
                     "created" => Faker::Time.between(DateTime.now - 2, DateTime.now) ,
                     "contact-email" => "bob@example.com" },
-                    "relationships": {
-                			"provider": {
-                				"data":{
-                					"type":"providers",
-                					"id": provider.symbol
+                    "relationships" => {
+                			"provider" => {
+                				"data" => {
+                					"type" => "providers",
+                					"id" => "ABC"
                 				}
                 			}
                 		}} }
   end
   let(:headers) { {'ACCEPT'=>'application/vnd.api+json', 'CONTENT_TYPE'=>'application/vnd.api+json', 'Authorization' => 'Bearer ' + User.generate_token}}
-  let(:query) { "jamon"}
 
-  # Test suite for GET /clients
-  describe 'GET /clients' do
-    before do
-      Provider.create(provider)         
-      clients.each {|client| post '/clients', params: client.to_jsonapi.to_json, headers: headers  }
-      # dois.each   { |item| Doi.create(item) }
-      sleep 2 
```

Comment:
ElasticSearch takes more than 1sec to index a resource.  So after create you might need to add a timeout.